drivers/I2C.cpp

```diff
@@ -120,6 +124,82 @@ void I2C::unlock() {
     _mutex->unlock();
 }
 
+int I2C::reset() {
+    lock();
+    // Try i2c_reset() i.e. target specific reset
+    i2c_reset(&_i2c);
+
+    // SCL and SDA as input pins with pullup
+    gpio_init_inout(&_gpio_scl, _scl, PIN_INPUT, PullUp, 1);
+    gpio_init_inout(&_gpio_sda, _sda, PIN_INPUT, PullUp, 1);
+
+    // Read and verify if reset was successful
+    if (gpio_read(&_gpio_scl)) {
+        if (gpio_read(&_gpio_sda)) {
+            // Return successfuly as SDA and SCL is high
+            unlock();
```

Comment:
I will correct that, but need some suggestions. I am not sure, if this reset should be part of constructor or init by default.  Or as standalone API? Initially I kept it as part of constructor, but than realised we don't handle failure cases in constructors.